tests/DockerFileGeneratorTest.php

```diff
@@ -25,15 +25,16 @@ class DockerFileGeneratorTest extends TestCase
     /**
      * @dataProvider generatorDataProvider
      */
-    public function test_it_can_generate_a_dockerfile_contents(
+    public function test_all_extension_names_are_in_dockerfile(
         string $image,
         array $extensions,
         string $sourcePhar,
-        string $expected,
-    ): void {
-        $actual = (new DockerFileGenerator($image, $extensions, $sourcePhar))->generateStub();
 
-        $this->assertSame($expected, $actual);
+    ): void {
+        $stub = (new DockerFileGenerator($image, $extensions, $sourcePhar))->generateStub();
+        foreach($extensions as $extension) {
+            self::assertStringContainsString($extension, $stub);
```

Comment:
I have mixed feelings about this: you no longer really testing the content of the Dockerfile anymore although it is less dynamic now

Reply:
I felt the template testing makes the test too coupled to the implementation. We could simply do an integration test that just builds the container and checks if each dependency is available.

Reply:
>We could simply do an integration test that just builds the container and checks if each dependency is available.

I think that would be really sweet and with this having a looser (but more flexible) test is perfectly fine